src/xcode/ENA/ENA/Source/Scenes/Vaccination/VaccinationQRCodeScannerViewController.swift

```diff
@@ -0,0 +1,250 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import AVFoundation
+
+class VaccinationQRCodeScannerViewController: UIViewController {
+	// MARK: - Init
+
+	init(
+		didScanCheckin: @escaping (String) -> Void,
+		dismiss: @escaping () -> Void
+	) {
+		self.dismiss = dismiss
+		
+		super.init(nibName: nil, bundle: nil)
+		
+		self.viewModel = VaccinationQRCodeScannerViewModel(
+			onSuccess: { [weak self] qrCodeString in
+				AudioServicesPlaySystemSound(SystemSoundID(kSystemSoundID_Vibrate))
+				self?.viewModel?.deactivateScanning()
+			//	didScanCheckin(qrCodeString)
+			},
+			onError: { error in
+				switch error {
+				// for the moment we always show the same alert
+				case .cameraPermissionDenied:
+					DispatchQueue.main.async {
+						self.dismiss()
+					}
+				default:
+					self.showErrorAlert(error: error)
```

Comment:
Shouldn't this also happen on main?

Reply:
what do you mean by main? if you mean the other cases then no, we show the permissions dialog and if the user rejected we just dismiss without showing an error

Reply:
I mean the main thread. In the other case, the UI is called explicitly on main. Could make sense here too.